apps/system/js/feature_detector.js

```diff
@@ -0,0 +1,55 @@
+/* global BaseModule */
+'use strict';
+
+(function(exports) {
+  /**
+   * This is the feature detector module of the system app.
+   * It is responsible to retrieve device specific information.
+   */
+  var FeatureDetector = function() {
+  };
+
+  FeatureDetector.STATES = [
+    'getDeviceType',
+    'getDeviceMemory'
+  ];
+
+  BaseModule.create(FeatureDetector, {
+    name: 'FeatureDetector',
+    deviceType: null,
+    deviceMemory: 0,
+
+    _start: function() {
+      this._getDeviceType();
+      this._getDeviceMemory();
+    },
+
+    _stop: function() {
+      this.deviceType = null;
+      this.deviceMemory = 0;
+    },
+
+    _getDeviceType: function() {
+      // `_GAIA_DEVICE_TYPE_` will be replaced by real device type in build time
```

Comment:
I prefer to add a comment pointing to apps/system/build/build.js who does this change in build time.